modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js

```diff
@@ -71,31 +84,29 @@ const BalloonEditor = ({config = {}, contents, name, ...otherProps}) => {
 
 				const balloonToolbars = editor.balloonToolbars;
 
-				balloonToolbars.create({
-					buttons: editorConfig.toolbarText,
-					cssSelector: '*',
-				});
-
-				balloonToolbars.create({
-					buttons: editorConfig.toolbarImage,
-					priority:
-						window.CKEDITOR.plugins.balloontoolbar.PRIORITY.HIGH,
-					widgets: 'image,image2',
-				});
-
-				balloonToolbars.create({
-					buttons: editorConfig.toolbarTable,
-					cssSelector: 'td',
-					priority:
-						window.CKEDITOR.plugins.balloontoolbar.PRIORITY.HIGH,
-				});
-
-				balloonToolbars.create({
-					buttons: editorConfig.toolbarVideo,
-					priority:
-						window.CKEDITOR.plugins.balloontoolbar.PRIORITY.HIGH,
-					widgets: 'videoembed',
-				});
+				const toolbarPanelConfigs = Object.fromEntries(
+					Object.entries(editorConfig).filter(([key, _]) =>
+						/^toolbar.*$/.test(key)
+					)
+				);
+
+				const toolbarPanelConfigsKeys = Object.keys(
+					toolbarPanelConfigs
+				);
+
+				if (toolbarPanelConfigsKeys.length > 0) {
+					toolbarPanelConfigsKeys.forEach((configKey) => {
+						if (editorConfig[configKey]) {
+							balloonToolbars.create({
+								buttons: editorConfig[configKey],
+								priority:
+									window.CKEDITOR.plugins.balloontoolbar
+										.PRIORITY.HIGH,
+								...TOOLBAR_PANEL_OTHER_CONFIGS[configKey],
+							});
+						}
+					});
+				}
```

Comment:
Uh, I think [the KISS principle](https://en.wikipedia.org/wiki/KISS_principle) applies here :joy: . I realize you removed some code duplication, but the code is a lot harder to read.

Can we have simple falsy checks? For example:
```js
editorConfig.toolbarImage && balloonToolbars.create({
	buttons: editorConfig.toolbarImage,
	priority: window.CKEDITOR.plugins.balloontoolbar.PRIORITY.HIGH,
	widgets: 'image,image2',
});
```

Reply:
> I realize you removed some code duplication, but the code is a lot harder to read and maintain.

Yes, in the end it became this 🤷🏼‍♂️ 😂 

> Can we have simple falsy checks?

It will look more simpler. Going in this direction 🏃🏼‍♂️

Reply:
When going with falsy checks(short circuits) using `&&` like:

```js
editorConfig.toolbarImage && balloonToolbars.create({
...
```

An error is thrown:

```log
~ckeditor! ❯ yarn format                                                                                            10.26s ~ckeditor
yarn run v1.22.4
warning package.json: No license field
$ liferay-npm-scripts fix
/Users/diego/code/portal/liferay-portal/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js
  74:5  error  Expected an assignment or function call and instead saw an expression.  no-unused-expressions
  80:5  error  Expected an assignment or function call and instead saw an expression.  no-unused-expressions
  89:5  error  Expected an assignment or function call and instead saw an expression.  no-unused-expressions
  98:5  error  Expected an assignment or function call and instead saw an expression.  no-unused-expressions

✖ 4 problems (4 errors, 0 warnings)
```

We can add an exception to our eslint rules to prevent by adding:

[`allowShortCircuit: true`](https://eslint.org/docs/rules/no-unused-expressions#options)

I'll go with simple conditionals

Reply:
See https://github.com/liferay-frontend/liferay-portal/pull/1344/commits/babe6980a51d37358e9ff7cafdb7fd298a30cbe6